db/active_replicator_common.go

```diff
@@ -302,14 +316,17 @@ func (a *activeReplicatorCommon) getLastError() error {
 	return a.lastError
 }
 
-func (a *activeReplicatorCommon) getStateWithErrorMessage() (state string, lastErrorMessage string) {
-	a.stateErrorLock.RLock()
-	defer a.stateErrorLock.RUnlock()
+func (a *activeReplicatorCommon) _getStateWithErrorMessage() (state string, lastErrorMessage string) {
```

Comment:
Can we note this function is only safe to call when stateErrorLock is acquired?

I think we can push this code as is, but I think I'd be likely to break this state into its own datastructure, so you can indicate when you are calling it like

`GetStateRLock() (string, string)`
`GetStateWithoutLock() (string, string)`

Reply:
Underscore and non-underscore variants of functions have prior-art but I do agree with noting _which_ lock is required.

e.g:
- `changeListener.CurrentCount()`
- `ServerContext.fetchAndLoadDatabase()`
- `CollectionBucket.GetCollectionID()`
- `ServerContext.isDatabaseSuspended()`